camel-core/src/main/java/org/apache/camel/model/ModelHelper.java

```diff
@@ -53,12 +70,38 @@ public static String dumpModelAsXml(CamelContext context, NamedNode definition)
             jaxbContext = context.getModelJAXBContextFactory().newJAXBContext();
         }
 
+        final Map<String, String> namespaces = new LinkedHashMap<>();
+
+        // gather all namespaces from the routes or route which is stored on the expression nodes
+        if (definition instanceof RoutesDefinition) {
+            List<RouteDefinition> routes = ((RoutesDefinition) definition).getRoutes();
+            for (RouteDefinition route : routes) {
+                extractNamespaces(route, namespaces);
+            }
+        } else if (definition instanceof RouteDefinition) {
+            RouteDefinition route = (RouteDefinition) definition;
+            extractNamespaces(route, namespaces);
+        }
+
         Marshaller marshaller = jaxbContext.createMarshaller();
         marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
         StringWriter buffer = new StringWriter();
         marshaller.marshal(definition, buffer);
 
-        return buffer.toString();
+        Document dom = context.getTypeConverter().convertTo(Document.class, buffer.toString());
+
+        // Add additional namespaces to the document root element
+        Element documentElement = dom.getDocumentElement();
+        for (String nsPrefix : namespaces.keySet()) {
+            documentElement.setAttribute("xmlns:" + nsPrefix, namespaces.get(nsPrefix));
+        }
+
+        Exchange exchange = ExchangeBuilder.anExchange(context).build();
```

Comment:
Very clever. But I would like to see a way to convert this without having to create a dummy exchange. Maybe you can add some methods on the xml converter to accept those indent options or something and call the converter directly.

Creating an exchange is really only intended for routing messages.